File: Go/slack.go
package main

import (
	"bytes"
	"log"
	"net/http"
)

func sendMessage(channel, text string) {
	url := "https://hooks.slack.com/services/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"
	jsonStr := `{
		"channel":"` + channel + `",
		"username":"bot",
		"text":"` + text + `",
		"icon_emoji": ":beer:"
	}`
	req, err := http.NewRequest(
		"POST",
		url,
		bytes.NewBuffer([]byte(jsonStr)),
	)
	if err != nil {
		log.Print(err)
	}
	req.Header.Set("Content-Type", "application/json")
	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		log.Print(err)
	}
	log.Print(resp)
	defer resp.Body.Close()
}

func main() {
	sendMessage("#random", "hello, from Golang")
}

File: Ruby/slack.rb
require 'net/http'
require 'uri'
require 'json'

def send_message(channel, text)
  uri = URI.parse('https://hooks.slack.com/services/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')
  params = {
    "channel": channel,
    "username": "Bot",
    "text": text,
    "icon_emoji": ":beer:"
  }

  https = Net::HTTP.new(uri.host, uri.port)
  https.use_ssl = true
  req = Net::HTTP::Post.new(uri.request_uri)
  req['Content-Type'] = 'application/json'
  payload = params.to_json
  req.body = payload
  res = https.request(req)
  
  puts "code -> #{res.code}"
  puts "body -> #{res.body}"
end

send_message('#random', 'hello, from ruby')
